stdlib/random.mli

```diff
@@ -147,6 +147,41 @@ module State : sig
       correlation.  Both PRNGs can be split later, arbitrarily many times.
       @since 5.0 *)
 
+  val to_binary_string : t -> string
+  (** Serializes the PRNG state into an immutable sequence of bytes.
+      See {!of_binary_string} for deserialization.
+
+      The [string] type is intended here for serialization only, the
+      encoding is not human-readable and may not be printable.
+
+      Note that the serialization format may differ across OCaml
+      versions.
+
+      @since 5.1
+  *)
+
+  val of_binary_string : string -> t
+  (** Deserializes a byte sequence obtained by {!to_binary_string}.
+      The resulting PRNG state should produce exactly the same random
+      numbers as the state that was passed in input to
+      {!to_binary_string}.
+
+      @raise Failure if the input is not in the expected format.
+
+      Note that the serialization format may differ across OCaml
+      versions.
+
+      Unlike the functions provided by the {!Marshal} module, this
+      function either produces a valid state or fails cleanly with
+      a [Failure] exception. It can be safely used on user-provided,
+      untrusted inputs.
+
+      It could be useful, for example, to store the current random
+      state in the save file of a video game, or to have players
+      communicate their random state over the network.
```

Comment:
I don't think we need this paragraph.

Reply:
@damiendoligez @dbuenzli, I would be curious to know just a little more about your impression here. I thought on the contrary that it was sensible to try to distill in the documentation the sort of use-cases that convinced us (me at least?) to add the feature in the first place. (People often mention that they want to see examples in the documentation. This is not a code example but a usage example, and I thought it could be useful.) I would be curious to hear your opinion on this point to inform my future documentation-writing practice.

Reply:
Maybe it's style but I don't like chatty reference docs. I find these kind of suggestions a bit noisy and in the way. This is the kind of stuff you rather have in tutorials/howto docs. 

I rarely peruse reference documentation in search of use-cases, I rather peruse it knowing what I'm looking for. As such I want docstrings to communicate me efficiently how to use the structure item, possible pitfalls, pointers to nudge me to alternatives (where task based suggestions are ok) or towards more extensive docs. Nothing more distracting.

Keeping doc strings to the point also enables more efficient scanning of a module's offering (this is the reason why I often defer via a link a long module preamble or extensive explanations to a section at the end of the module)